### PR TITLE
CDK module changes for destination

### PR DESCRIPTION
**Pull request by gisripa**
### TL;DR
Updated JDBC and S3 destination configurations.

### CODE REVIEW

**Comment by gisripa** [comment]
* **#36589** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36589?utm_source=stack-comment-icon" target="_blank">Graphite</a>
* **#36588** <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36588?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @gisripa and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/DestinationAcceptanceTest.kt [hunk 1]
```diff
@@ -90,7 +90,7 @@ abstract class DestinationAcceptanceTest {
     protected var localRoot: Path? = null
     open protected var _testDataComparator: TestDataComparator = getTestDataComparator()
 
-    open fun getTestDataComparator(): TestDataComparator {
+    open protected fun getTestDataComparator(): TestDataComparator {
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/DestinationAcceptanceTest.kt:93]: I hope we have some formatter rules in Kotlin land to fix these kind of mismatches .. `protected open` vs `open protected`

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt [hunk 2]
```diff
@@ -38,7 +38,7 @@ object TestingNamespaces {
      *
      * @return convention-compliant namespace
      */
-    @JvmOverloads
+    @JvmStatic
     fun generate(prefix: String? = null): String {
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt:42]: Generated decided to use `@JvmOverloads` for static overloaded methods. Maybe a Companion object with default params and `@JvmOverloads` might be better here ?

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3StorageOperations.kt [hunk 3]
```diff
@@ -416,6 +416,10 @@ open class S3StorageOperations(
         )
     }
 
+    fun uploadManifest(bucketName: String, manifestFilePath: String, manifestContents: String) {
+        s3Client.putObject(s3Config.bucketName, manifestFilePath, manifestContents)
+    }
+
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3StorageOperations.kt:422]: This method got nuked out of orbit.. diff is just resurrection.

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/io/IOs.kt [hunk 4]
```diff
@@ -13,6 +13,7 @@ import java.util.*
 import org.apache.commons.io.input.ReversedLinesFileReader
 
 object IOs {
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/io/IOs.kt:15]: This whole class should be just companion object.

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/jackson/MoreMappers.kt [hunk 5]
```diff
@@ -16,6 +16,7 @@ import com.fasterxml.jackson.datatype.jsr310.JavaTimeModule
  * All jackson mapper creation should use the following methods for instantiation.
  */
 object MoreMappers {
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/jackson/MoreMappers.kt:18]: This whole class should be just companion object.

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/json/Jsons.kt [hunk 6]
```diff
@@ -44,9 +44,10 @@ object Jsons {
     private val YAML_OBJECT_MAPPER: ObjectMapper = MoreMappers.initYamlMapper(YAMLFactory())
```
Comment (gisripa) [inline airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/json/Jsons.kt:44]: <i class='graphite__hidden'>[Re: line 29]<br/><br/></i>

This whole class should be just companion object. <span class='graphite__hidden'><br/><br/>See this comment inline on <a href="https://app.graphite.dev/github/pr/airbytehq/airbyte/36588?utm_source=unchanged-line-comment">Graphite</a>.</span>

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt (hunk 2 above)
Comment (jdpgrailsdev) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt:42]: I think that the Kotlin convention would be to use the companion object, but I don't believe that there is really any technical difference.  I think its more about readability and is really only temporary as long as we have Java code that needs to call this/will go away when that code becomes Kotlin too.

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/validation/json/JsonSchemaValidator.kt [hunk 7]
```diff
@@ -7,7 +7,7 @@ import com.fasterxml.jackson.databind.JsonNode
 import com.google.common.annotations.VisibleForTesting
 import com.google.common.base.Preconditions
 import com.networknt.schema.*
-import io.airbyte.commons.string.Strings.join
+import io.airbyte.commons.string.Strings.Companion.join
```
Comment (jdpgrailsdev) [inline airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/validation/json/JsonSchemaValidator.kt:10]: nit for our cleanup:  Kotlin has a `joinToString` extension function.

**Review by jdpgrailsdev — APPROVED** [review]
:ship:

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt (hunk 2 above)
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt:42]: the @JvmOverload says that the compiler will generate 2 static `generate` functions: one with no parameter, and one with 1 parameter. Has little to do with static.
You really want `@JvmOverload @JvmStatic`

Comment (gisripa) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt:42]: For this method to be used in java's static context like `TestingNamespaces.generate()` && `TestingNamespace.generate("string")` had to use `@JvmStatic`. `@JvmOverloads` works with instance methods. Its a different story if used in Kotlin since kotlin supports `TestingNamespaces.generate()` call on singleton `object`

Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt:42]: I actually tested this. If the function `generate(prefix: String? = null): String` is annotated with `@JvmStatic` and `@JvmOverloads`, and the function `generate()` is removed, then the compiled will create a JVM `static public generate()` function. Let's do that

Comment (gisripa) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/TestingNamespaces.kt:42]: Interesting. double annotate you mean. Will try that.

File: airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/DestinationAcceptanceTest.kt (hunk 1 above)
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/db-destinations/src/testFixtures/kotlin/io/airbyte/cdk/integrations/standardtest/destination/DestinationAcceptanceTest.kt:93]: I think there's a compiler warning against that. Right now I had to disable them but will eventually re-enable them as errors

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/io/IOs.kt (hunk 4 above)
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/io/IOs.kt:15]: this is not a class, this is an object. As such, it's its own companion object. So everything in it can be declared @JvmStatic

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/jackson/MoreMappers.kt (hunk 5 above)
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/jackson/MoreMappers.kt:18]: see above

File: airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/string/Strings.kt [hunk 8]
```diff
@@ -6,18 +6,20 @@ package io.airbyte.commons.string
 import java.util.*
 import org.apache.commons.lang3.RandomStringUtils
 
-object Strings {
-    fun join(iterable: Iterable<Any>, separator: CharSequence): String {
-        return iterable.joinToString(separator) { it.toString() }
-    }
+class Strings {
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/dependencies/src/main/kotlin/io/airbyte/commons/string/Strings.kt:9]: object Strings is the canonical way to do that. You should revert this change, and make the methods @JvmStatic

File: airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3StorageOperations.kt (hunk 3 above)
Comment (stephane-airbyte) [reply airbyte-cdk/java/airbyte-cdk/s3-destinations/src/main/kotlin/io/airbyte/cdk/integrations/destination/s3/S3StorageOperations.kt:422]: oops

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/ColumnId.kt [hunk 9]
```diff
@@ -14,18 +14,8 @@ package io.airbyte.integrations.base.destination.typing_deduping
  * Useful if a destination warehouse handles columns ignoring case, but preserves case in the table
  * schema.
  */
-class ColumnId(name: String, originalName: String, canonicalName: String) {
+data class ColumnId(val name: String, val originalName: String, val canonicalName: String) {
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/ColumnId.kt:17]: thank you

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/InitialRawTableStatus.kt [hunk 10]
```diff
@@ -6,18 +6,8 @@ package io.airbyte.integrations.base.destination.typing_deduping
...
 import java.util.*
 
-class InitialRawTableStatus(
-    rawTableExists: Boolean,
-    hasUnprocessedRecords: Boolean,
-    maxProcessedTimestamp: Optional<Instant>
-) {
-    val rawTableExists: Boolean
-    val hasUnprocessedRecords: Boolean
+data class InitialRawTableStatus(
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/main/kotlin/io/airbyte/integrations/base/destination/typing_deduping/InitialRawTableStatus.kt:9]: thank you

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.kt [hunk 11]
```diff
@@ -58,7 +58,7 @@ abstract class BaseSqlGeneratorIntegrationTest<DestinationState : MinimumDestina
     protected var cdcIncrementalAppendStream: StreamConfig? = null
 
     protected var generator: SqlGenerator? = null
-    protected abstract var destinationHandler: DestinationHandler<DestinationState>
+    protected abstract val destinationHandler: DestinationHandler<DestinationState>
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.kt:61]: not sure why it was defined as a var... Much better as a val. How did you find it?

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt [hunk 12]
```diff
@@ -1066,9 +1067,17 @@ abstract class BaseTypingDedupingTest {
         return Companion.readMessages(filename, streamNamespace, streamName)
     }
 
+    protected fun readRecords(filename: String): List<JsonNode> {
+        return Companion.readRecords(filename)
+    }
+
+    protected fun getSchema(): JsonNode {
```
Comment (stephane-airbyte) [inline airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseTypingDedupingTest.kt:1074]:
```
protected open val schema: JsonNode
   get = SCHEMA

File: airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.kt (hunk 11 above)
Comment (gisripa) [reply airbyte-cdk/java/airbyte-cdk/typing-deduping/src/testFixtures/kotlin/io/airbyte/integrations/base/destination/typing_deduping/BaseSqlGeneratorIntegrationTest.kt:61]: making var generated abstract `setDestinationHandler` method which was never intended to be, just `getDestinationHandler`

**Review by stephane-airbyte — APPROVED** [review]
